### PR TITLE
feat(deps): use RHDH themes in the backstage app and dev pages

### PR DESCRIPTION
**Pull request by jerolimov**
**Fixes:** https://issues.redhat.com/browse/RHIDP-1171

Use the RHDH theme (extracted as npm package) also in the backstage app and the dev pages.

For now it shows different themes, incl. backstage, RHDH 1.0, 1.1, and 1.2.

* https://github.com/redhat-developer/red-hat-developer-hub-theme
* https://npmjs.com/package/@redhat-developer/red-hat-developer-hub-theme

### CODE REVIEW

**Review by ciiay — APPROVED** [review]
LGTM
Tested working great.
Thank you 👏

**Comment by invincibleJai** [comment]
@jerolimov the test jobs seem to fail due to link issues

```
       @redhat-developer/red-hat-developer-hub-theme must be declared in devDependencies of plugins/analytics-module-matomo/package.json, run 'yarn --cwd plugins/analytics-module-matomo add --dev @redhat-developer/red-hat-developer-hub-theme' from the project root
  
  
       plugins/analytics-module-matomo/dev/index.tsx:5:1
       3 | import { createDevApp } from '@backstage/dev-utils';
       4 | 
     > 5 | import { createDevAppThemes } from '@redhat-developer/red-hat-developer-hub-theme';
         | ^
       6 | 
       7 | import { analyticsModuleMatomoPlugin } from '../src';
       8 | import { Playground } from './Playground';
  
  ✘ 1 problem (1 error, 0 warnings)
```

https://github.com/janus-idp/backstage-plugins/actions/runs/8833415969/job/24371757247?pr=1480#step:3:512

**Comment by ciiay** [comment]
Hi @jerolimov , the theme works perfectly on plugins. Just the theme options aren't scrollable.
rhdip_1171

**Comment by jerolimov** [comment]
> @jerolimov the test jobs seem to fail due to link issues
> 
> ```
>        @redhat-developer/red-hat-developer-hub-theme must be declared in devDependencies of plugins/analytics-module-matomo/package.json, run 'yarn --cwd plugins/analytics-module-matomo add --dev @redhat-developer/red-hat-developer-hub-theme' from the project root
>   
>   
>        plugins/analytics-module-matomo/dev/index.tsx:5:1
>        3 | import { createDevApp } from '@backstage/dev-utils';
>        4 | 
>      > 5 | import { createDevAppThemes } from '@redhat-developer/red-hat-developer-hub-theme';
>          | ^
>        6 | 
>        7 | import { analyticsModuleMatomoPlugin } from '../src';
>        8 | import { Playground } from './Playground';
>   
>   ✘ 1 problem (1 error, 0 warnings)
> ```
> 
> https://github.com/janus-idp/backstage-plugins/actions/runs/8833415969/job/24371757247?pr=1480#step:3:512

I fixed this kind of build errors. I believe that the other test errors aren't related to this PR. :man_shrugging:

**Comment by ciiay** [comment]
LGTM
image

**Comment by caponetto** [comment]
I haven't seen any issues with the Orchestrator plugin UX.
My only observation is that `overflowX` for Backstage page root is `scroll` not `auto` and kinda light mode (see the bottom bar in the screenshot). 

image

**Comment by karthikjeeyar** [comment]
Tested **Argocd** plugin and **Quay** plugin, both looks good and no issues found! 🎉 

image

**Comment by karthikjeeyar** [comment]
In **Tekton plugin**, **RHDH Light (latest)** and **RHDH Dark (latest)** is showing `--` under the collapsable rows . Other parts in tekton like logs, output dashboards etc looks good.

image

File: plugins/acr/package.json [hunk 1]
```diff
@@ -45,6 +45,7 @@
     "@backstage/dev-utils": "1.0.31",
     "@backstage/test-utils": "1.5.4",
     "@janus-idp/cli": "1.8.6",
+    "@redhat-developer/red-hat-developer-hub-theme": "*",
```
Comment (karthikjeeyar) [inline plugins/acr/package.json:48]: @jerolimov wouldnt' this pick any matched version, why not pin the version ?

Comment (jerolimov) [reply plugins/acr/package.json:48]: @karthikjeeyar , good question. I pinned one in `packages/app/package.json`. Pinning different per plugins doesn't work nicely and I don't wanted that we need to touch all packages.json for each theme update. With this we can just update `packages/app` and the `yarn.lock` of course.

Comment (karthikjeeyar) [reply plugins/acr/package.json:48]: Okay, but I dont see this package.json packages/app in this PR or in main branch, am I missing something? 🤔

Comment (jerolimov) [reply plugins/acr/package.json:48]: You're right, I missed it, maybe it's dropped in a rebase conflict. It's there again.

**Comment by jerolimov** [comment]
@karthikjeeyar @ciiay I've updated to include 3 merged PRs from @ciiay in the theme repo:

- https://github.com/redhat-developer/red-hat-developer-hub-theme/pull/8
- https://github.com/redhat-developer/red-hat-developer-hub-theme/pull/10
- https://github.com/redhat-developer/red-hat-developer-hub-theme/pull/11
- PR 8 fixes https://issues.redhat.com/browse/RHIDP-2289 - [UI] Hide scrollbar when not scrollable
- PR 10 fixes https://issues.redhat.com/browse/RHIDP-2290 - [UI] Elements located by "makeStyles-" classname doesn't get applied to elements on ci builds
- PR 11 fixes https://issues.redhat.com/browse/RHIDP-2300 - [UI] Remove double dash on unexpected components

Can you please take a look and try to get this in? And, of course, we will update the theme later regularly.

**Review by karthikjeeyar — APPROVED** [review]
Tested it works fine.

/lgtm

**Pull request merged**